compiler/rustc_parse/src/lexer/mod.rs

```diff
@@ -456,9 +456,13 @@ impl<'a> StringReader<'a> {
         };
         let content_start = start + BytePos(prefix_len);
         let content_end = suffix_start - BytePos(postfix_len);
-        let id = self.symbol_from_to(content_start, content_end);
-        self.validate_literal_escape(mode, content_start, content_end, prefix_len, postfix_len);
-        (lit_kind, id)
+        let has_fatal_err =
+            self.validate_literal_escape(mode, content_start, content_end, prefix_len, postfix_len);
+        if !has_fatal_err {
+            (lit_kind, self.symbol_from_to(content_start, content_end))
+        } else {
+            (token::Err, self.symbol_from_to(start, suffix_start))
```

Comment:
Is it necessary to produce a `token::Err` here rather than `LitKind::Err` in `fn from_token_lit`?
It seems like the latter would be preferrable.

Reply:
Also, any specific reasons to make the span different in the error and non-error cases?

Reply:
These changes were necessary to get back the original error messages for the tests where my earlier version had changed the messages.

For your first suggestion: that would require detecting the unescaping problem here but then delaying the reporting to `from_token_lit`, which doesn't seem right.

As for the spans: for valid quoted literals we don't include the quotes in the symbol. Except in the error case, otherwise we end up printing something like
```
bad literal `foo`
```
instead of:
```
bad literal `"foo"`
```
The latter is better and what the tests currently do.